sdk/java/src/main/java/net/consensys/gpact/functioncall/CrossControlManagerGroup.java

```diff
@@ -0,0 +1,98 @@
+package net.consensys.gpact.functioncall;
+
+import java.util.ArrayList;
+import net.consensys.gpact.common.BlockchainConfig;
+import net.consensys.gpact.common.BlockchainId;
+import net.consensys.gpact.messaging.MessagingVerificationInterface;
+import org.web3j.crypto.Credentials;
+
+/**
+ * Crosschain Manager Group implementations provide the SDK with an ability to manage Crosschain
+ * Control Contracts across multiple blockchains.
+ */
+public interface CrossControlManagerGroup {
+
+  /**
+   * This method should be only used in testing situations. Importantly, it deploys a new Crosschain
+   * Control Contract.
+   *
+   * <p>Configure a blockchain for use by the function call layer.
+   *
+   * @param creds Credentials to be used for signing transactions on the specified blockchain.
+   * @param bcInfo Information about the blocchain: Blockchain id, URL, gas pricing strategy.
+   * @param messageVerification Implementation used to obtain signed events or proofs of events so
+   *     that the events can be used on remote blockchains.
+   * @throws Exception If an issue is encountered deploying the crosschain control contract.
+   */
+  void addBlockchainAndDeployContracts(
```

Comment:
I don't see the value if you keep adding contract deployment code to a Java SDK directly. This code should be on deployment scripts that are common in whatever language you have. 

Duplicating this for each SDK is not scalable.

I started a new small project using the GPACT contracts and see how I can set up the the examples read/write using Hardhat+TS.

Reply:
I think that different users will want to do things differently. Especially for testing / getting things working initially, I can imagine users wanting to be able to deploy an entire system and run some scenario all in the one executable.  Once the deployment scripts have been created, maybe this API could call to the deployment script.